### ISSUE TITLE
Bug: rotation is opposite on Android and iOS

### ISSUE THREAD

**Issue by ben-xD** [liodali/osm_flutter#355]
When we ask osm_flutter to rotate by 90 degrees, 

- On Android, it rotates clockwise
- On iOS, it rotates anti-clockwise

If we rotate the camera by -90 degrees (90 degrees clockwise), this is equivalent to the **map** rotating the opposite way, 90 degrees (90 degrees anticlockwise).

Because the API on osm_flutter says "rotate_map", on iOS we should invert the rotation, because the API does mapView.setCameraPosition.

```swift
    private func rotateMap(call: FlutterMethodCall) {
        let angle = call.arguments as! Double
        if (angle > 0.0) {
            mapView.setCameraPosition(TGCameraPosition(center: mapView.position, zoom: mapView.zoom, bearing: CLLocationDirection(CGFloat(angle)), pitch: 0.0), withDuration: 0.2, easeType: TGEaseType.sine)
        }
    }
```

On Android, the API rotates the **map**:

from [[docs](https://osmdroid.github.io/osmdroid/javadocAll/org/osmdroid/views/MapController.html)](https://osmdroid.github.io/osmdroid/javadocAll/org/osmdroid/views/MapController.html)

> Start animating the map towards the given point.

```kotlin
    private fun mapOrientation(call: MethodCall, result: MethodChannel.Result) {
        //map!!.mapOrientation = (call.arguments as Double?)?.toFloat() ?: 0f
        // Convert rotation into orientation
        val orientation = (call.arguments as Double?)?.toFloat() ?: 0f
        map!!.controller.animateTo(
                map!!.mapCenter,
                map!!.zoomLevelDouble,
                null,
                orientation
        )
        mapSnapShot().saveMapOrientation(map!!.mapOrientation)
        map!!.invalidate()
        result.success(null)
    }
```

**TLDR:** RotateMap != Rotate camera

---

**Side note:** Mapbox doesn't have **map** rotation API, but **camera** rotation. 

---

## My workaround

It's pretty easy to fix this on my end, but I think it would be good to fix this for everyone.

```
      if (Platform.isIOS) {
        final rotationCompensation = -yaw;
        final positiveOnlyRotationCorrection = (rotationCompensation < 0)
            ? rotationCompensation + 360
            : rotationCompensation;
        mapController.rotateMapCamera(positiveOnlyRotationCorrection);
      } else if (Platform.isAndroid) {
        final rotationCompensation = yaw;
        // OSM on Android rotates the **wrong way.** I will make an PR for this.
        mapController.rotateMapCamera(rotationCompensation);
      }
```

---

Side note, I got the error with the example app, so I just had to update the deployment target to be the same for debug, release and profile in Xcode:

```
RuntimeError - [Xcodeproj] Consistency issue: build setting `IPHONEOS_DEPLOYMENT_TARGET` has multiple
    values: `{"Debug"=>"11.0", "Release"=>"11.0", "Profile"=>"13.0"}`
```

**Comment by ben-xD** [comment]
I would probably fix it on Android: because the android API is for rotating the **map**, but the OSM API says rotate camera. So on Android, the angle needs to be reversed.

I can also contribute an MR if you want :)

**Comment by liodali** [comment]
yeah you can make pr to fix that for sure
